gym/envs/robotics/hand/manipulate_touch_sensors.py

```diff
@@ -0,0 +1,146 @@
+            touch_get_obs (string): touch sensor readings
+                - "boolean": returns 1 if touch sensor reading != 0.0 else 0
+                - "log": returns log(x+1) touch sensor readings from self.sim.data.sensordata[]
+                - "off" or else: does not add touch sensor readings to the observation
+
+        """
+        self.touch_visualisation = touch_visualisation
+        self.touch_get_obs = touch_get_obs
+        self._tsensor_prefix = 'robot0:TS_'
+        # self._tsensor_group_names = []  # list of dictionaries of sensors per joint
+        # self._tsensor_name2id = {}
+        self._tsensor_id2name = {}
+        self._tsensor_id2siteid = {}
+        self._site_id2intial_rgba = {}  # dict for initial rgba values for debugging
+        self.touch_color = [1, 0, 0, 0.4]  # [1, 0, 0, 0.4]
+        self.notouch_color = [0, 0.5, 0, 0.3]  # [0, 0.5, 0, 0.1]
+
+        manipulate.ManipulateEnv.__init__(
+            self, model_path, target_position, target_rotation,
+            target_position_range, reward_type, initial_qpos=initial_qpos,
+            randomize_initial_position=randomize_initial_position, randomize_initial_rotation=randomize_initial_rotation,
+            distance_threshold=distance_threshold, rotation_threshold=rotation_threshold, n_substeps=n_substeps, relative_control=relative_control,
+            ignore_z_target_rotation=ignore_z_target_rotation,
+        )
+        utils.EzPickle.__init__(self)
+
+        for k, v in self.sim.model._sensor_id2name.items():  # get touch sensor ids and their site names
+            if 'TS' in v:
+                # self._tsensor_name2id[v] = k
+                self._tsensor_id2name[k] = v
+                self._tsensor_id2siteid[k] = self.sim.model._site_name2id[v.replace('TS', 'T')]
+                self._site_id2intial_rgba[self._tsensor_id2siteid[k]] = self.sim.model.site_rgba[self._tsensor_id2siteid[k]].copy()  # get initial rgba values
+
+
+    def _get_obs(self):
+        robot_qpos, robot_qvel = manipulate.robot_get_obs(self.sim)
+        object_qvel = self.sim.data.get_joint_qvel('object:joint')
+        achieved_goal = self._get_achieved_goal().ravel()  # this contains the object position + rotation
+        touch_values = []  # get touch sensor readings. if there is one, set value to 1
+        if self.touch_get_obs == 'boolean':
+            for k, v in self._tsensor_id2name.items():
+                value = 1.0 if self.sim.data.sensordata[k] != 0.0 else 0.0
+                touch_values.append(value)
+        elif self.touch_get_obs == 'log':
+            for k, v in self._tsensor_id2name.items():
+                value = self.sim.data.sensordata[k]
+                touch_values.append(value)
+            if len(touch_values) > 0:
+                touch_values = np.log(np.array(touch_values) + 1.0)
+        observation = np.concatenate([robot_qpos, robot_qvel, object_qvel, touch_values, achieved_goal])
+
+        # set rgba values
```

Comment:
Please move everything related to visualization to `_render_callback` (https://github.com/openai/gym/blob/master/gym/envs/robotics/robot_env.py#L152) instead. Don't forget to call `super` since the hand environment itself does some visualization here as well.